### PR TITLE
feat: added dynamic imports for web components

### PR DESCRIPTION
**Pull request by sstefdev**
### Changes:
 - Added dynamic imports to web-components
 
 Previous performance:
 - Mobile: 32
 - Desktop: 62
 
 This is updated performance:

- Mobile: Image
- Desktop: Image

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic imports for the `CreateInvoiceForm` and `InvoiceDashboard` components, enhancing loading performance.
	- Added loading placeholders that display a spinner while components are being fetched.
	- Added new `Spinner` component for loading animations.
	- Expanded available components with `Navbar` and `VersionDisplay`.

- **Bug Fixes**
	- Improved component rendering by disabling server-side rendering for both components, ensuring a smoother user experience.

### CODE REVIEW

File: pages/create-invoice.tsx [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+import dynamic from "next/dynamic";
 import { config } from "@/utils/config";
 import { useAppContext } from "@/utils/context";
 import { currencies } from "@/utils/currencies";
 import { rainbowKitConfig as wagmiConfig } from "@/utils/wagmiConfig";
-import CreateInvoiceForm from "@requestnetwork/create-invoice-form/react";
+
+const CreateInvoiceForm = dynamic(
+  () => import("@requestnetwork/create-invoice-form/react"),
+  { ssr: false, loading: () => <p>Loading...</p> }
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply pages/create-invoice.tsx:10]: Please use a loading indicator (spinner) instead of a loading message.

File: pages/index.tsx [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+import dynamic from "next/dynamic";
 import { config } from "@/utils/config";
 import { useAppContext } from "@/utils/context";
 import { currencies } from "@/utils/currencies";
 import { rainbowKitConfig as wagmiConfig } from "@/utils/wagmiConfig";
-import InvoiceDashboard from "@requestnetwork/invoice-dashboard/react";
+
+const InvoiceDashboard = dynamic(
+  () => import("@requestnetwork/invoice-dashboard/react"),
+  { ssr: false, loading: () => <p>Loading...</p> }
```
Comment (MantisClone) [inline pages/index.tsx:10]: Same as previous comment.

**Review by MantisClone — APPROVED** [review]
Approved contingent on comment resolution

**Comment by sstefdev** [comment]
Added spinner

Screenshot 2024-10-22 at 09 24 33

File: components/common/index.ts [hunk 3]
```diff
@@ -1,2 +1,4 @@
 export { default as Button } from "./Button";
 export { default as Dropdown } from "./Dropdown";
+export { default as Navbar } from "./Navbar";
+export { default as VersionDisplay } from "./VersionBadge";
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply components/common/index.ts:4]: No need for this

File: pages/create-invoice.tsx [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
 import { useAppContext } from "@/utils/context";
 import { currencies } from "@/utils/currencies";
 import { rainbowKitConfig as wagmiConfig } from "@/utils/wagmiConfig";
-import CreateInvoiceForm from "@requestnetwork/create-invoice-form/react";
+import { Spinner } from "@/components/ui";
+
+const CreateInvoiceForm = dynamic(
+  () => import("@requestnetwork/create-invoice-form/react"),
+  { ssr: false, loading: () => <Spinner /> }
+);
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply pages/create-invoice.tsx:12]: No need

**Pull request merged**